File: go/available_parameters.go
// this file is @generated
package meteroid

type AvailableParameters struct {
	// Map of component_id -> available billing periods (e.g., "MONTHLY", "ANNUAL")
	BillingPeriods map[string][]BillingPeriodEnum `json:"billing_periods,omitempty"`

	// Map of component_id -> available capacity values
	CapacityThresholds map[string][]int64 `json:"capacity_thresholds,omitempty"`

	// List of component_ids that support slot parametrization (initial slot count)
	SlotComponents []string `json:"slot_components,omitempty"`
}

File: python/meteroid/models/available_parameters.py
# this file is @generated
import dataclasses
import typing as t

from ..serialization import BaseModel
from .billing_period_enum import BillingPeriodEnum


@dataclasses.dataclass
class AvailableParameters(BaseModel):
    billing_periods: t.Optional[t.Dict[str, t.List[BillingPeriodEnum]]] = None
    """Map of component_id -> available billing periods (e.g., "MONTHLY", "ANNUAL")"""

    capacity_thresholds: t.Optional[t.Dict[str, t.List[int]]] = None
    """Map of component_id -> available capacity values"""

    slot_components: t.Optional[t.List[str]] = None
    """List of component_ids that support slot parametrization (initial slot count)"""
